### PR TITLE
upload files

### PR DESCRIPTION
**Pull request by nikolicstjepan**

### CODE REVIEW

**Comment by cloverstudio** [comment]
This part in file.ts could be problem when file size is huge.
```
async function checkHashes(hash: string, filePath: string) {
    if (!hash) {
        return true;
    }
    const content = await readFile(filePath, "utf-8");

    const fileHash = Utils.sha1(content);
    return hash === fileHash;
}
```
Can you try to change this part to use readable stream ?
https://stackoverflow.com/questions/18658612/obtaining-the-hash-of-a-file-using-the-stream-capabilities-of-crypto-module-ie

**Comment by cloverstudio** [comment]
Please add a test with real hash value from a file, you can use test.png.

File: test/07_upload_files.ts [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+                .post("/api/upload/files")
+                .set({ accesstoken: globals.userToken })
+                .send({ ...validParams, relationId: 5 });
+
+            expect(responseInvalidNotNumber.status).to.eqls(400);
+            expect(responseNoParam.status).to.eqls(200);
+            expect(responseValid.status).to.eqls(200);
+        });
+
+        it("Returns error if hash is not matching", async () => {
```
Comment (nikolicstjepan) [inline test/07_upload_files.ts:284]: @kenyasue tu je taj test

Comment (cloverstudio) [reply test/07_upload_files.ts:284]: Ali bih htio da radiš baš test sa pravi file, jer ovako ne testira se slučaj kad se spaja file. To je dosta bitno jer zna biti problema kod spajanja.

**Review by cloverstudio — APPROVED** [review]
LGTM

**Pull request merged**